### PR TITLE
Add log archive storage

### PR DESCRIPTION
**Pull request by tei-k**
# Description
https://scalar-labs.atlassian.net/browse/DLT-7006

# Done
- Add feature for log archive to cloud storage (S3 and Azure blob)
- The default is not to archive.
- The s3 bucket must be created in the same region as other resources.
- Just support AWS -> S3, Azure -> Azure blob

### CODE REVIEW

File: modules/azure/monitor/locals.tf [hunk 1]
```diff
@@ -57,4 +58,9 @@ locals {
     local.monitor_base[var.base],
     var.monitor
   )
+
+  log_archive_storage = regex("^(?:https://(?P<storage_account>[^.]+)[^/]+(?:/(?P<container_name>[^.]+)))?", local.monitor.log_archive_storage_base_uri)
+
```
Comment (feeblefakie) [inline modules/azure/monitor/locals.tf:63]: Can we remove the empty line? (for consistency)

File: modules/universal/monitor/vars.tf [hunk 2]
```diff
@@ -68,5 +68,10 @@ variable "targets" {
 
 variable "log_retention_period_days" {
   default     = 30
-  description = "Set the retention period of the aggregated log to the monitor server"
+  description = "Set the retention period of the aggregated log to the monitor host"
```
Comment (feeblefakie) [inline modules/universal/monitor/vars.tf:71]:
```suggestion
  description = "Set the retention period of the accumulated logs in the monitor host"
```

File: modules/universal/monitor/vars.tf [hunk 3]
```diff
@@ -68,5 +68,10 @@ variable "targets" {
...
+}
+
+variable "log_archive_storage_info" {
+  default     = ""
+  description = "A info to set archive storage of the aggregated log to the monitor host"
```
Comment (feeblefakie) [inline modules/universal/monitor/vars.tf:76]:
```suggestion
  description = "An info to set archive storage of the accumulated logs in the monitor host"
```

It is not related to this PR but I think all the descriptions are a bit inconsistent in the sense that some start with a noun describing what it is and others start with a verb describing what it does.

**Review by feeblefakie — COMMENTED** [review]
Overall looking good! Left minor comments.

File: modules/azure/monitor/locals.tf (hunk 1 above)
Comment (tei-k) [reply modules/azure/monitor/locals.tf]: OK! Fixed in 2e5c892.

File: modules/universal/monitor/vars.tf (hunk 2 above)
Comment (tei-k) [reply modules/universal/monitor/vars.tf]: Fixed in ff60449.

File: modules/universal/monitor/vars.tf (hunk 3 above)
Comment (tei-k) [reply modules/universal/monitor/vars.tf]: Fixed in ff60449.

File: modules/azure/monitor/identity.tf [hunk 4]
```diff
@@ -0,0 +1,32 @@
+data "azurerm_subscription" "subscription" {}
+
+data "azurerm_storage_account" "monitor" {
+  count = local.enable_log_archive_storage && local.monitor.resource_count > 0 ? 1 : 0
+
+  resource_group_name = local.network_name
+  name                = local.log_archive_storage.storage_account
+}
+
+resource "azurerm_role_definition" "monitor" {
```
Comment (ymorimo) [inline modules/azure/monitor/identity.tf:10]: This is good compared to Cassy's identity that uses the built-in `Storage Blob Data Contributor` role.

**Review by ymorimo — APPROVED** [review]
LGTM!

**Pull request merged**